### PR TITLE
[Datastore] Improve `BigQuerySource` documentation to cover credentials

### PR DESCRIPTION
**Pull request by gtopper**
Related to [ML-3745](https://jira.iguazeng.com/browse/ML-3745)

### CODE REVIEW

File: docs/store/datastore.md [hunk 1]
```diff
@@ -112,6 +112,7 @@ authentication methods that use the `fsspec` mechanism.
...
 * `GOOGLE_APPLICATION_CREDENTIALS` &mdash; path to the application credentials to use (in the form of a JSON file). This can
 be used if this file is located in a location on shared storage, accessible to pods executing MLRun jobs.
-* `GCP_CREDENTIALS` &mdash; when the credentials file cannot be mounted to the pod, this environment variable may contain
-the contents of this file. If configured in the function pod, MLRun dumps its contents to a temporary file 
-and points `GOOGLE_APPLICATION_CREDENTIALS` at it.
\ No newline at end of file
+* `GCP_CREDENTIALS` &mdash; when the credentials file cannot be mounted to the pod, this secret or environment variable
+may contain the contents of this file. If configured in the function pod, MLRun dumps its contents to a temporary file 
+and points `GOOGLE_APPLICATION_CREDENTIALS` at it. An exception is `BigQuerySource`, which uses `GCP_CREDENTIALS`
+directly.
```
Comment (theSaarco) [inline docs/store/datastore.md:118]:
```suggestion
and points `GOOGLE_APPLICATION_CREDENTIALS` at it. An exception is `BigQuerySource`, which passes `GCP_CREDENTIALS` contents
directly to the BigQuery client.
```

Comment (gtopper) [reply docs/store/datastore.md:118]: Actually, it passes the contents to Spark.

Comment (gtopper) [reply docs/store/datastore.md:118]: Ah, and to the client in pandas engine.

Comment (gtopper) [reply docs/store/datastore.md:118]:
```suggestion
and points `GOOGLE_APPLICATION_CREDENTIALS` at it. An exception is `BigQuerySource`, which passes `GCP_CREDENTIALS`'s
contents directly to the BigQuery client or, when using spark engine, to Spark.
directly.
```
?

Comment (theSaarco) [reply docs/store/datastore.md:118]: That works, though it may be a bit TMI - I'm fine with it.

Comment (gtopper) [reply docs/store/datastore.md:118]: Could be "directly to the query engine". Wdyt?

**Review by assaf758 — APPROVED** [review]

**Pull request merged**